gradle-jdks/src/main/java/com/palantir/gradle/jdks/ToolchainsPlugin.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.jdks;
+
+import com.palantir.baseline.plugins.javaversions.BaselineJavaVersionsExtension;
+import com.palantir.baseline.plugins.javaversions.ChosenJavaVersion;
+import com.palantir.gradle.jdks.GradleJdkConfigs.GradleJdkConfigsTask;
+import com.palantir.gradle.jdks.GradleWrapperPatcher.GradleWrapperPatcherTask;
+import java.util.List;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.tasks.TaskProvider;
+import org.gradle.api.tasks.wrapper.Wrapper;
+import org.gradle.language.base.plugins.LifecycleBasePlugin;
+
+public final class ToolchainsPlugin extends JdkDistributionConfigurator implements Plugin<Project> {
```

Comment:
Feels weird to extend `JdkDistributionConfigurator` when it just has static methods - I'd just call the statics as normal.